packages/documentsearch/src/searchoverlay.tsx

```diff
@@ -702,7 +706,7 @@ namespace Private {
     queryString: string,
     caseSensitive: boolean,
     regex: boolean
-  ) {
+  ): RegExp {
```

Comment:
The regular expression triggering CodeQL below was discussed in: https://github.com/jupyterlab/jupyterlab/issues/7881#issuecomment-892196340

Reply:
Thanks for the cross reference